erpnext/stock/doctype/purchase_receipt/purchase_receipt.py

```diff
@@ -284,9 +284,14 @@ def make_item_gl_entries(self, gl_entries, warehouse_account=None):
 		for d in self.get("items"):
 			if d.item_code in stock_items and flt(d.valuation_rate) and flt(d.qty):
 				if warehouse_account.get(d.warehouse):
-					stock_value_diff = frappe.db.get_value("Stock Ledger Entry",
+					stock_value_diff = frappe.db.get_values("Stock Ledger Entry",
 						{"voucher_type": "Purchase Receipt", "voucher_no": self.name,
-						"voucher_detail_no": d.name, "warehouse": d.warehouse, "is_cancelled": 0}, "stock_value_difference")
+						"voucher_detail_no": d.name, "warehouse": d.warehouse, "is_cancelled": 0}, "stock_value_difference", as_dict=True, pluck="stock_value_difference")
+					has_serial = frappe.db.get_value('Item', d.item_code, 'has_serial_no')
+					if has_serial:
+						stock_value_diff = sum(stock_value_diff)
+					else:
+						stock_value_diff = stock_value_diff[0]
```

Comment:
It can always be the sum IMO. Make it future-proof. 

Alternatively `sum(stock_value_difference)` can be fetched directly.